doc/wcmp/WCMP-test-plan.md

```diff
@@ -0,0 +1,295 @@
+![Use case scenario](images/WCMP_enabled.png)
+
+
+## Related DUT CLI commands
+
+### Config
+The following command can be used to enable/disable WCMP:
+```
+config bgp device-global wcmp enabled
+config bgp device-global wcmp disabled
+```
+
+### Show
+The following command can be used to show WCMP status:
+```
+show bgp device-global
+show bgp device-global --json
+```
+Example:
+```
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+```
+admin@router:~$ show bgp device-global --json
+{
+    "tsa": "disabled",
+    "wcmp": "enabled"
+}
+```
+
+
+## Related DUT configuration files
+```
+"BGP_DEVICE_GLOBAL": {
+    "STATE": {
+        "tsa_enabled": "false",
+        "wcmp_enabled": "true"
+    }
+}
+```
+
+
+## Test cases
+### Test case #1 - WCMP command line test
+1. Enable WCMP and verify WCMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp enabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+2. Verify BGP link bandwidth extended community is added in FRR
+```
+router# show running-config bgp
+Building configuration...
+
+Current configuration:
+!
+route-map TO_BGP_PEER_V4 permit 100
+ set extcommunity bandwidth num-multipaths
+exit
+!
+route-map TO_BGP_PEER_V6 permit 100
+ set extcommunity bandwidth num-multipaths
+exit
+!
+end
+```
+3. Verify the route weight value on remote host is 25, which means WCMP is effect
+```
+admin@router:~$ ip route show 100.0.0.1
+100.0.0.1 proto bgp metric 20
+        nexthop via 10.10.26.1 dev Ethernet100 weight 25
+        nexthop via 10.10.27.1 dev Ethernet104 weight 25
+        nexthop via 10.10.28.1 dev Ethernet108 weight 25
+        nexthop via 10.10.29.1 dev Ethernet112 weight 25
+```
+4. Disable WCMP and verify WCMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp disabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  disabled
+```
+5. Verify BGP link bandwidth extended community is deleted in FRR
+```
+router# show running-config bgp
+Building configuration...
+
+Current configuration:
+!
+route-map TO_BGP_PEER_V4 permit 100
+exit
+!
+route-map TO_BGP_PEER_V6 permit 100
+exit
+!
+end
+```
+6. Verify the route weight value on remote host is 1, which means WCMP is disabled
+```
+admin@router:~$ ip route show 100.0.0.1
+100.0.0.1 proto bgp metric 20
+        nexthop via 10.10.26.1 dev Ethernet100 weight 1
+        nexthop via 10.10.27.1 dev Ethernet104 weight 1
+        nexthop via 10.10.28.1 dev Ethernet108 weight 1
+        nexthop via 10.10.29.1 dev Ethernet112 weight 1
+```
+7. Config WCMP with invalid parameter, verify config fail and report error message
+```
+admin@router:~$ sudo config bgp device-global wcmp 1
+Usage: config bgp device-global wcmp [OPTIONS] COMMAND [ARGS]...
+Try "config bgp device-global wcmp -h" for help.
+
+Error: No such command "1".
+```
+
+### Test case #2 - Config WCMP from config_db.json file
```

Comment:
@weiguo-nvidia no need to repeat the same twice: this test case only makes sense to catch an error message produced by a daemon when invalid configuration is provided

Reply:
Remove this case